vmware/global-load-balancing-services-for-kubernetes
# Service Discovery: Integration test suite

Service Discovery: Integration test suite

Coverage:
- Two member clusters and one management cluster
- MCI validation
- Each member cluster with one service each
- Service creation -> ServiceImport creation verification
- Update node port in service -> ServiceImport endpoints update
  verification
- Update port in service (port not present in MCI) -> ServiceImport
  deletion verification
- Update service type from NodePort to ClusterIP -> ServiceImport
  deletion verification (as only NodePort services are supported for
  now)
- Update service type from ClusterIP to NodePort -> ServiceImport is
  created back
- Delete node in a cluster -> ServiceImport endpoint update verification
- Add node in a cluster -> ServiceImport endpoint update verification
- Delete service in both clusters -> ServiceImport objects deletion

Couple of fixes:
- Service type change from NodePort to ClusterIP had no effect which
  is not desired. Fixed in layer 2 to check the service against
  accepted types and then delete it if a service import previously
  existed for this backend service.
- Bootup handler had a sequence which checked for MCI objects first
  and then initialized the clusterset filter. This would cause the
  filter to not be initialized if no MCI objects were added. Fix is to
  reverse this sequence.
